lib/widgets/action_sheet.dart

```diff
@@ -366,3 +368,34 @@ class CopyButton extends MessageActionSheetMenuItemButton {
       data: ClipboardData(text: rawContent));
   }
 }
+
+class MarkAsUnread extends MessageActionSheetMenuItemButton {
+  MarkAsUnread({
+    super.key,
+    required super.message,
+    required super.messageListContext,
+    required this.model,
+  });
+
+  final MessageListView model;
+
+  @override IconData get icon => Icons.mark_chat_unread_outlined;
+
+  @override
+  String label(ZulipLocalizations zulipLocalizations) {
+    return zulipLocalizations.actionSheetMarkAsUnread;
+  }
+
+  @override void onPressed(BuildContext context) async {
+    Navigator.of(context).pop();
+    final ids = [
+      message.id,
+      ...model.getNarrowMessagesAfter(message).map((e) => e.id)];
```

Comment:
Is there a standard (cleaner) way to get messages that are in the same narrow and come after a specific message?